### ISSUE TITLE
Billing Address field in AddDebitCardPage doesn't accept text

### ISSUE THREAD

**Issue by hayata-suenaga** [Expensify/App#14918]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Break down in numbered steps

1. Go to Settings > Payments
2. Tap Add payment method > Debit card
3. Try to type in Billing Address field

## Expected Result:

Inputted text should be displayed.

## Actual Result:

Text doesn't appear when text is inputted.

## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

I tried to paste a text in the field instead of typing, but it doesn't work either. No workaround found.

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation
**Expensify/Expensify Issue URL:**
**Issue reported by:**
**Slack conversation:** https://expensify.slack.com/archives/C03TQ48KC/p1675793090919109

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~011c6f32902f1a74c9</li>
        <li>Upwork Job ID: 1623035041406087168</li>
        <li>Last Price Increase: 2023-02-07</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mallenexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Triggered auto assignment to @grgia (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by aldo-expensify** [comment]
Removing @grgia because @hayata-suenaga is working on it and I think he already found the cause

**Comment by hayata-suenaga** [comment]
`maxInputLength` prop that is passed to `TextInput` inside `AddressSearch` component seems to the the root cause.

**Comment by mountiny** [comment]
Can you confirm what versions this is happening on, we just had a deploy so I want to know if this is in production now or the fresh new staging. In which case it would not be a deploy blocker.

**Comment by aldo-expensify** [comment]
> Can you confirm what versions this is happening on, we just had a deploy so I want to know if this is in production now or the fresh new staging. In which case it would not be a deploy blocker.

@mountiny I tested in production and it is not there

**Comment by aldo-expensify** [comment]
@mountiny I tested with version v1.2.65-2

**Comment by mountiny** [comment]
Ok yeah so that is production now so this does not have to be a deploy blocker! But I think we should still CP to Staging the fix so this gets out sooner

**Comment by mountiny** [comment]
The staging right not is https://github.com/Expensify/App/issues/14911 1.2.67

**Comment by situchan** [comment]
I confirmed that #14290 caused regression.
Especially these:
https://github.com/Expensify/App/pull/14290/files#diff-ad5309d68bcb873fce55c9426aa6103b2d0e5ea58334e34be2dca40d33b4dad0R86-R87
https://github.com/Expensify/App/pull/14290/files#diff-ad5309d68bcb873fce55c9426aa6103b2d0e5ea58334e34be2dca40d33b4dad0R232
Native apps don't treat -1 as unlmited

**Comment by aldo-expensify** [comment]
This is the PR that caused the regression: https://github.com/Expensify/App/pull/14290 and seems like it is only in staging for the moment.

> @mountiny I tested with version v1.2.65-2

I wasn't clear, that is the production version I saw. Retested production, I still see the version `v1.2.65-2` and the bug is not there.

**Comment by mountiny** [comment]
I see, thanks for the context! And thanks for working on the quick fix!

dont forget to add the CP Staging label before you merge the PR

**Comment by aldo-expensify** [comment]
@parasharrajat ready for review now

**Comment by aldo-expensify** [comment]
Merged the PR

**Comment by hayata-suenaga** [comment]
Can I close the issue now?

**Comment by aldo-expensify** [comment]
> Can I close the issue now?

It will automatically close once it is deployed (I think), in most cases you don't need to close it manually

**Comment by hayata-suenaga** [comment]
Because the fix has been deployed, I'll go ahead and close this issue.

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.2.67-7 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/14921


If no regressions arise, payment will be issued on 2023-02-15. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [] Propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mallenexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by mallenexpensify** [comment]
Gonna propose this for the TestRail Test case/regression update. 

Add new test under "Account Settings" 
Title "Payments - add card details"
Step 1 - Go to Settings > Payments
2. Tap Add payment method > Debit card
3. Try to type in Billing Address field
4. Verify Inputted text is displayed.

**Comment by aldo-expensify** [comment]
Going to self assign to help with the CME checklist

**Comment by mallenexpensify** [comment]
From #qa, Applause doesn't have debit card credentials for testing, it's being worked on https://github.com/Expensify/Expensify/issues/226985  so I put this issue on hold pending that. 

Also, per @aldo-expensify in [the Slack thread](https://expensify.slack.com/archives/C01SKUP7QR0/p1675968185579419?thread_ts=1675910956.175229&cid=C01SKUP7QR0), fleshing out the steps makes sense here

1. Go to Settings > Payments
2. Tap Add payment method > Debit card
3. Try to type 123 in Billing Address field
4. Verify Inputted text is displayed.
5. Verify that you get some suggestions in a dropdown
6. Click one of the suggestions
7. Verify that Billing address, Zip code and State get updated
8. Use the testing debit card Name/Number: xxx/yyy
9. Save
10. Verify that the debit card was added

**Comment by hayata-suenaga** [comment]
This issue is being held until the QA can have a test debit card to implement the above testing steps.

**Comment by aldo-expensify** [comment]
Issue about testing debit card credential for QA: https://github.com/Expensify/Expensify/issues/226985

**Comment by hayata-suenaga** [comment]
I thought if the issue title has [HOLD], melvin-bot will be chill. does anyone know if we created a test card for QA team to use?

**Comment by mountiny** [comment]
No, HOLD does not stop melvin, only Reviewing does, but that should only be used for actual PR being in a review.

You can also change the priority of the issue if appropriate

**Comment by aldo-expensify** [comment]
I think Applause has the ability to test with cards now: https://github.com/Expensify/Expensify/issues/226985#issuecomment-1502170174

@hayata-suenaga he should be able to proceed with updating the test: https://github.com/Expensify/App/issues/14918#issuecomment-1424963464

**Comment by mallenexpensify** [comment]
Dropped in #qa to ask about getting this tested 
https://expensify.slack.com/archives/C9YU7BX5M/p1681317318570329

**Comment by hayata-suenaga** [comment]
> @hayata-suenaga he should be able to proceed with updating the test: https://github.com/Expensify/App/issues/14918#issuecomment-1424963464

Has the comment already been updated?

**Comment by hayata-suenaga** [comment]
@aldo-expensify nvm I was confused between updating testing steps in the original comment and actual Applause tests 😅

**Comment by mallenexpensify** [comment]
@mvtglobally can you test this?  

Here's Maria's reply in #QA
> we can add Debit card now. Only issue tester faced when set the newly added debit card as the default payment method - Default label is NOT present ( Step 10 -  Add debit card section).  Team logged two issue related default payment https://github.com/Expensify/App/issues/16911 and https://github.com/Expensify/App/issues/16914
Let me know if I need to log a separate issue for Debit Card.

**Comment by mvtglobally** [comment]
I will update in the morning with details

**Comment by mvtglobally** [comment]
@Beamanator @mallenexpensify 
Doesn't look like this issue is repro any more https://user-images.githubusercontent.com/43995119/232060261-e274c8e9-3622-4dbc-a8be-6eb656a33bf6.MOV
https://user-images.githubusercontent.com/43995119/232060295-21b051c4-55c2-476d-845a-9179e5ce50d6.MOV
https://user-images.githubusercontent.com/43995119/232060336-21a3c7ea-f6b8-4029-a67e-ee1457daa6fa.MOV

**Comment by mallenexpensify** [comment]
Thanks @mvtglobally , I'm going to close then.  Comment/reopen if needed.

**Comment by MelvinBot** [comment]
@mallenexpensify @aldo-expensify @hayata-suenaga  Be sure to fill out the [Contact List](https://forms.gle/zhdenCB5dE5gKfnw6)!
